qubesagent/vmexec.py

```diff
@@ -0,0 +1,56 @@
+import sys
+import os
+import re
+
+
+class DecodeError(ValueError):
+    pass
+
+
+def decode_part(part):
+    if not re.match(r'^[a-zA-Z0-9._-]*$', part):
+        raise DecodeError('illegal characters found')
+
+    part = part.encode('ascii')
+
+    part = part.replace(b'--', b'/')
```

Comment:
This could use a comment, especially why it's safe.

Reply:
Right, it's actually not safe :( Things like `-2Fbin-2Fls` get converted to `/bin/ls` then replaced to `-bin-ls`. I found a better way.